### PR TITLE
Update missing_ids.json with 2024 rookie GSIS data

### PR DESCRIPTION
**Pull request by mcarman8**
There are no GSIS ids for 2024 rookies in the master ff_playerids database, this adds the GSIS ids for every rookie on a 53-man roster and most rookie free agents represented in MFL. A similar thing was done in https://github.com/dynastyprocess/data/pull/50 last year.
The only 3 rookies not represented are Taulia Tagovailoa, Miyan Williams, and Zakhari Franklin.

For future reference, the code used for this update is:
```
library(nflreadr)
library(tidyverse)
library(dplyr)
library(jsonlite)

#Get all rookies from 2024 and 2024 with MFL_ids
ff_playerids <- nflreadr::load_ff_playerids() %>%
  dplyr::filter(draft_year >= 2024)

#Get all Rookies from 2024 and their gsis_ids
rosters_weekly_2024 <- nflreadr::load_rosters_weekly(seasons = 2024) %>%
  dplyr::filter(rookie_year >= 2024) %>%
  dplyr::distinct(gsis_id, .keep_all = TRUE) %>%
  dplyr::select(team, full_name, gsis_id) %>%
  dplyr::rename(name=full_name)

#Update Team Names that Didn't Match
rosters_weekly_2024 <- rosters_weekly_2024 %>%
  dplyr::mutate(team = replace(team, team == "NE", "NEP")) %>%
  dplyr::mutate(team = replace(team, team == "TB", "TBB")) %>%
  dplyr::mutate(team = replace(team, team == "NO", "NOS")) %>%
  dplyr::mutate(team = replace(team, team == "GB", "GBP")) %>%
  dplyr::mutate(team = replace(team, team == "KC", "KCC")) %>%
  dplyr::mutate(team = replace(team, team == "LV", "LVR")) %>%
  dplyr::mutate(team = replace(team, team == "SF", "SFO")) %>%
  dplyr::mutate(team = replace(team, team == "LA", "LAR")) %>%
  dplyr::mutate(team = replace(team, team == "JAX", "JAC"))

#Remove names with extra Jr., III, etc. and set to lowercase for merge names
rosters_weekly_2024$name <- sub("^(\\S*\\s+\\S+).*", "\\1", rosters_weekly_2024$name)
rosters_weekly_2024$name <- tolower(rosters_weekly_2024$name)
library(stringr)
rosters_weekly_2024$name <- str_replace_all(rosters_weekly_2024$name, fixed("."), "")
rosters_weekly_2024$name <- str_replace_all(rosters_weekly_2024$name, fixed("'"), "")

#At least 1 player has a name that doesn't match, fix that individually
rosters_weekly_2024 <- rosters_weekly_2024 %>%
  dplyr::mutate(name = replace(name, name == "mike hall", "michael hall")) %>%
  dplyr::mutate(name = replace(name, name == "audric estimé", "audric estime")) %>%
  dplyr::mutate(name = replace(name, name == "levi drake", "levi drake-rodriguez"))

#Lideatrick Griffin hasn't been updated to the browns, so force the update
ff_playerids <- ff_playerids %>% 
  mutate(team = ifelse(team == "LVR" & merge_name == "lideatrick griffin", "CLE", team)) %>%
  mutate(team = ifelse(team == "NYJ" & merge_name == "brandon codrington", "BUF", team))


#Rename to merge name
rosters_weekly_2024 <- rosters_weekly_2024 %>%
  dplyr::rename(merge_name=name)


combined <- ff_playerids %>% 
  dplyr::select(-c(gsis_id)) %>%
  dplyr::left_join(rosters_weekly_2024, by=join_by(merge_name)) # Removed Team join to include all Free Agents in case they get signed

#3 players do not get matches: Taulia Tagovailoa, Miyan Williams, and Zakhari Franklin
combined <- combined %>% drop_na(gsis_id)

output_dataframe <- combined %>%
  dplyr::select(mfl_id, gsis_id)

output_dataframe <- transform(output_dataframe, mfl_id = as.numeric(mfl_id))

json_output <- toJSON(output_dataframe) %>% prettify()

write(json_output, "missing_players_update.json")
```

### CODE REVIEW

File: .Rhistory [hunk 1]
```diff
@@ -0,0 +1,3 @@
+db_playerids <- readRDS("~/Fantasy Football/dynastyprocess_data/files/db_playerids.rds")
```
Comment (tanho63) [inline .Rhistory]: can you remove this .Rhistory file? should be `git rm .Rhistory`

Comment (mcarman8) [reply .Rhistory]: Done. I didn't realize that was in there, thanks.

**Comment by tanho63** [comment]
Is `espn_id_new` from your PR or from #56? is it safe to coalesce them?

**Comment by mcarman8** [comment]
> Is `espn_id_new` from your PR or from #56? is it safe to coalesce them?

`espn_id_new` is from #56. I was unaware of it because I just auto-merged our missing_ids.json files. I'm assuming that would have no effect on the results? Or should I update it to be espn_id?

**Comment by tanho63** [comment]
Yeah I'd prefer to coalesce them, I think it's an artifact of the cleanup process

**Pull request merged**